### PR TITLE
On Windows, fix window size for maximized, undecorated windows

### PR DESCRIPTION
**Pull request by msiglreith**
Mostly takes the relevant pieces from https://github.com/rust-windowing/winit/pull/1891 regarding max window handling which haven't been included so far.
Attributed @amrbashir in the commit accordingly, I hope that's fine!

Fixes https://github.com/rust-windowing/winit/issues/2568

- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

File: src/platform_impl/windows/event_loop.rs [hunk 1]
```diff
@@ -978,19 +978,27 @@ unsafe fn public_window_callback_inner<T: 'static>(
...
-            // ahead of the window surface. Currently, there seems no option to achieve this
-            // with the Windows API.
-            if window_flags.contains(WindowFlags::MARKER_UNDECORATED_SHADOW) {
-                let params = &mut *(lparam as *mut NCCALCSIZE_PARAMS);
+            let params = &mut *(lparam as *mut NCCALCSIZE_PARAMS);
+
+            if util::is_maximized(window) {
+                // Limit the window size when maximized to the current monitor.
+                // Otherwise it would include the non-existent decorations.
+                let monitor = monitor::current_monitor(window);
```
Comment (amrbashir) [inline src/platform_impl/windows/event_loop.rs:986]: using `monitor::current_monitor` which uses `MonitorFromWindow` and `MONITOR_DEFAULTTONEAREST` is unreliable in this situation, see https://github.com/MicrosoftEdge/WebView2Feedback/issues/2549#issuecomment-1180843867, instead, `MonitorFromRect` should be used

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you for the attribution, unfortunately a bug with this code came to light since then, see below.

**Comment by kchibisov** [comment]
@msiglreith is this good to go?

**Comment by msiglreith** [comment]
not yet, I will update it today

**Pull request merged**